extensions/controllers/sandboxclaim_controller.go

```diff
@@ -341,3 +356,167 @@ func (r *SandboxClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+						if port.Protocol != "" {
+							proto = port.Protocol
+						}
+						ingressPorts = append(ingressPorts, networkingv1.NetworkPolicyPort{
+							Protocol: &proto,
+							Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: p},
+						})
+					}
+				}
+			}
+			ingressRules = append(ingressRules, networkingv1.NetworkPolicyIngressRule{
+				From:  ingressPeers,
+				Ports: ingressPorts,
+			})
+		}
+
+		var egressRules []networkingv1.NetworkPolicyEgressRule
+		dnsPort53 := intstr.FromInt(53)
+		protoUDP := corev1.ProtocolUDP
+		protoTCP := corev1.ProtocolTCP
+		egressRules = append(egressRules, networkingv1.NetworkPolicyEgressRule{
+			To: []networkingv1.NetworkPolicyPeer{{
+				NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "kube-system"}},
+				PodSelector:       &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": "kube-dns"}},
+			}},
+			Ports: []networkingv1.NetworkPolicyPort{
+				{Protocol: &protoUDP, Port: &dnsPort53},
+				{Protocol: &protoTCP, Port: &dnsPort53},
+			},
+		})
+
+		for _, rule := range templateNP.AdditionalEgressRules {
+			var egressPeers []networkingv1.NetworkPolicyPeer
+			var egressPorts []networkingv1.NetworkPolicyPort
+			if rule.ToIPBlock != nil {
+				egressPeers = append(egressPeers, networkingv1.NetworkPolicyPeer{
+					IPBlock: &networkingv1.IPBlock{CIDR: rule.ToIPBlock.CIDR, Except: rule.ToIPBlock.Except},
+				})
+			} else if rule.ToPodSelector != nil {
+				peer := networkingv1.NetworkPolicyPeer{}
+				if rule.InNamespaceSelector != nil {
+					peer.NamespaceSelector = &metav1.LabelSelector{MatchLabels: rule.InNamespaceSelector}
+				}
+				peer.PodSelector = &metav1.LabelSelector{MatchLabels: rule.ToPodSelector}
+				egressPeers = append(egressPeers, peer)
+			}
+			for _, p := range rule.Ports {
+				portNum := intstr.FromInt(int(*p.Port))
+				proto := corev1.ProtocolTCP
+				if p.Protocol != nil {
+					proto = *p.Protocol
+				}
+				egressPorts = append(egressPorts, networkingv1.NetworkPolicyPort{
+					Protocol: &proto,
+					Port:     &portNum,
+				})
+			}
+			egressRules = append(egressRules, networkingv1.NetworkPolicyEgressRule{
+				To:    egressPeers,
+				Ports: egressPorts,
+			})
+		}
+
+		np.Spec = networkingv1.NetworkPolicySpec{
+			PodSelector: podSelector,
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+			Ingress: ingressRules,
+			Egress:  egressRules,
+		}
+
+		return controllerutil.SetControllerReference(claim, np, r.Scheme)
+	})
+
+	if err != nil {
+		logger.Error(err, "Failed to create or update NetworkPolicy for claim")
+		return err
+	}
+
+	logger.Info("Successfully reconciled NetworkPolicy for claim", "NetworkPolicy.Name", np.Name)
+	return nil
+}
+
+// NameHash generates an FNV-1a hash from a string and returns
+// it as a fixed-length hexadecimal string.
+func NameHash(objectName string) string {
```

Comment:
Is it possible to reuse the same method in the SandboxController?  https://github.com/kubernetes-sigs/agent-sandbox/blob/main/controllers/sandbox_controller.go#L236